server/usercache.go

```diff
@@ -88,10 +88,14 @@ func (luc *ldapUserCache) Update() error {
 		userKeys := []ssh.PublicKey{}
 		for _, eachKey := range entry.GetAttributeValues("sshPublicKey") {
 			sshKeyBytes, _ := base64.StdEncoding.DecodeString(eachKey)
-			userSSHKey, err := ssh.ParsePublicKey(sshKeyBytes)
-			if err != nil {
-				log.Errorf("SSH key parsing for %s failed! This key will not be added into LDAP.", username)
-				continue
+			userSSHKey, err1 := ssh.ParsePublicKey(sshKeyBytes)
+			if err1 != nil {
+				userSSHKey2, _, _, _, err2 := ssh.ParseAuthorizedKey([]byte(eachKey))
```

Comment:
userSSHKey2 and err2 feel a little awkward in there. At that point you don't really need the old error nor userSSHKey, so you can reuse them and make the code a bit cleaner. Something like:
```userSSHKey, _, _, _, err = ssh.ParseAuthorizedKey([]byte(eachKey))```